### ISSUE TITLE
Decorated method doesn't seem to type check correctly

### ISSUE THREAD

**Issue by drosseau** [microsoft/pyright#9847]
The following code doesn't type check correctly:

```python
from typing import Callable
import functools


def invoke[**P](f: Callable[P, None], *args: P.args, **kwargs: P.kwargs):
    print(f"args = {args}, kwargs = {kwargs}")
    f(*args, **kwargs)


def wrapper[**P](f: Callable[P, None]):
    @functools.wraps(f)
    def _wrapper(*args: P.args, **kwargs: P.kwargs):
        invoke(f, *args, **kwargs)

    return _wrapper


class Greeter:
    def __init__(self, greeting: str):
        self.greeting = greeting

    @wrapper
    def greet(self, name: str):
        print(f"{self.greeting} {name}")


g = Greeter("Hello")
g.greet("world")
```

```
$ pyright test.py
/tmp/test/test.py
  /tmp/test/test.py:28:1 - error: Argument missing for parameter "name" (reportCallIssue)
1 error, 0 warnings, 0 informations
```

```
$ python test.py
args = (<__main__.Greeter object at 0x7ff2182d9940>, 'world'), kwargs = {}
Hello world
```

```
$ pyright --version
pyright 1.1.393
```

**Comment by drosseau** [comment]
Sorry I didn't realize there was a bug in the code :)

**Comment by benglewis** [comment]
@drosseau What was the bug? I don't see anything and I do see the type check error from pyright and it seems quite similar to an issue that I am facing while using Google Cloud Storage's Client... :/

**Comment by drosseau** [comment]
I missed your question. If anyone comes across this again, the `wrapper` didn't have a return type.

```
def wrapper[**P](f: Callable[P, None]) -> Callable[P, None]:
    @functools.wraps(f)
    def _wrapper(*args: P.args, **kwargs: P.kwargs):
        invoke(f, *args, **kwargs)

    return _wrapper
```

This typechecks correctly.
